### ISSUE TITLE
Renamed: git:init during composer install uses wrong path

### ISSUE THREAD

**Issue by kvdnberg** [phpro/grumphp#145]
When (re-)installing GrumPHP using composer install or update inside vagrant, the pre-commit hook looks like this: 
```vagrant ssh --command "(exec '/vagrant/bin/grumphp' 'git:pre-commit' '--skip-success-output')"```

This gives an exception running git outside of vagrant: 
```
 [Gitonomy\Git\Exception\LogicException]
  Can't create a working copy on a bare repository
```
When reinitializing using ```bin/grumphp git:init``` inside vagrant, the line is changed to:
```vagrant ssh --command "(exec 'bin/grumphp' 'git:pre-commit' '--skip-success-output')"```

**Comment by veewee** [comment]
The `bin/grumphp` command is the correct one. When exactly did you get the `/vagrant` in your git hook?
I'm not fond of adding new parameters to the `grumphp.yml` specific for the vagrant hook.
It should be universal so that new hook presets can be added without altering the code. This is to keep the code as general as possible and independent of the file system or used tools.

That is why we decided that the line in .basrc is sufficient. You could add this file to the vagrant provision so that every user has the same .bashrc file. We put it in our boxes by default, so that's not a big problem. 

So in my oppinion the PR is not a valid fix. The issue that needs to be fixed is the hook with the `/vagrant` path and maybe a more clear section in the documentation.

**Comment by kvdnberg** [comment]
Thanks for your reply. I understand your objection to the fix.

The problem is still that it would seem that the bashrc line doesn't work for ```vagrant ssh --command```

```
$ vagrant ssh --command "pwd"
/home/vagrant
Connection to 127.0.0.1 closed.
$ vagrant ssh
vagrant@boxname:/vagrant #$ pwd
/vagrant
```

**Comment by kvdnberg** [comment]
Updated the ticket to reflect the /vagrant bug during composer install/update

**Comment by veewee** [comment]
Strange .. I'll have to dive deeper in the issue.
Thanks for reporting!

**Comment by veewee** [comment]
I've started debugging the git:init command during both composer install and when the command runs stand-alone.
After some digging, I found out that composer is adding the absolute path to the bin dir in the PATH environemnt variable during the dispatching of events:
https://github.com/composer/composer/blob/1.1/src/Composer/EventDispatcher/EventDispatcher.php#L147-L160

Since we are using the Symfony `ExecutableFinder`:
https://github.com/symfony/process/blob/master/ExecutableFinder.php, the absolute path to the executable is found before it reaches our relative `$extraDir` bin-dir.

Maybe we should create a custom ExecutableFinder that falls back on the paths in the PATH environment variable. The problem is that the ExecutableFinder is pretty neat and works on most OS, so it would be 99% copy-paste. Another option is to place the relative path before any other paths, but that feels a bit hacky. Both solutions  would also fix the issue #140.
